app/code/Magento/Catalog/Test/Unit/Block/Product/View/AttributesTest.php

```diff
@@ -0,0 +1,187 @@
+    {
+        $this->phraseResource = new Phrase( // @codingStandardsIgnoreLine
+            self::TESTPHRASETEXTISBLANK
+        );
+
+        $this->attributeMock = $this
+            ->getMockBuilder(AbstractAttribute::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getIsVisibleOnFront')
+            ->will($this::returnValue(true));
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getAttributeCode')
+            ->will($this::returnValue(self::TESTATTRIBUTENAME));
+
+        $this->frontendAttributeMock = $this
+            ->getMockBuilder(AbstractFrontend::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getFrontendInput')
+            ->willReturn(self::TESTATTRIBUTENAME);
+
+        $this->attributeMock
+            ->expects($this::any())
+            ->method('getFrontend')
+            ->willReturn($this->frontendAttributeMock);
+
+        $this->productMock = $this
+            ->getMockBuilder(Product::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->productMock
+            ->expects($this::any())
+            ->method('getAttributes')
+            ->willReturn([$this->attributeMock]);
+
+        $this->productMock
+            ->expects($this::any())
+            ->method('hasData')
+            ->willReturn(true);
+
+        $this->contextMock = $this
+            ->getMockBuilder(Context::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->registryMock = $this
+            ->getMockBuilder(Registry::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->registryMock
+            ->expects($this::any())
+            ->method('registry')
+            ->willReturn($this->productMock);
+
+        $this->priceCurrencyInterfaceMock = $this
+            ->getMockBuilder(PriceCurrencyInterface::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+
+        $this->attributesBlock = new AttributesBlock( // @codingStandardsIgnoreLine
+            $this->contextMock,
+            $this->registryMock,
+            $this->priceCurrencyInterfaceMock
+        );
+    }
+
+    /**
+     * @return void
+     */
+    public function testGetAttributesBooleanNoValue()
+    {
+        $this->phraseResource = new Phrase( // @codingStandardsIgnoreLine
+            self::TESTPHRASETEXTISBLANK
+        );
+
+        $this->frontendAttributeMock
+            ->expects($this::any())
+            ->method('getValue')
+            ->willReturn($this->phraseResource);
+
+        $attributes = $this->attributesBlock->getAdditionalData();
+
+        $this::assertTrue(empty($attributes[self::TESTATTRIBUTENAME]));
```

Comment:
Replace all static calls with `$this->assert*`.

Reply:
I modified these to static because I was receiving a warning for calling a static method dynamically, but it works regardless of which way they are called so I will swap them back.

Reply:
Warning in IDE? Surely it works, just that it is used as `$this->assert*` in most of the tests (~200 VS ~8000 occurrences).

Just found out that `$this->` is a recommended by PHPUnit author way: https://github.com/sebastianbergmann/phpunit/issues/1922, https://phpunit.de/manual/current/en/appendixes.assertions.html#appendixes.assertions.static-vs-non-static-usage-of-assertion-methods

Please eliminate these static calls.

Reply:
They have been swapped back in the latest commit